src/agentlab/llm/chat_api.py

```diff
@@ -227,23 +226,18 @@ def __call__(self, messages: list[dict]) -> dict:
         ):
             tracking.TRACKER.instance(input_tokens, output_tokens, cost)
 
-        return AIMessage(content=completion.choices[0].message.content)
+        return dict(role="assistant", content=completion.choices[0].message.content)
 
     def invoke(self, messages: list[dict]) -> dict:
         return self(messages)
 
 
 class OpenRouterChatModel(ChatModel):
-    def __init__(
-        self,
-        model_name,
-        api_key=None,
-        temperature=0.5,
-        max_tokens=100,
-    ):
+    def __init__(self, model_name, api_key=None, temperature=0.5, max_tokens=100, max_retry=1):
```

Comment:
it seems like all inits are the same. Perhaps the base class should do the init
and then it would call _post_init() or something if we need to setup something
specific here.